internal/testutils/parse_url.go

```diff
@@ -0,0 +1,51 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package testutils
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+)
+
+// Assume url.Parse may fail when rawTarget represents IPV4 or IPV6.
+func parseTargetAsIP(rawTarget string, urlParseError error) *url.URL {
+	ip := net.ParseIP(rawTarget) // TODO (easwars): should this be added to grpc.parseTarget?
+	if ip == nil {
+		panic(fmt.Sprintf("Error parsing target(%s): %v", rawTarget, urlParseError))
+	}
+	return &url.URL{
+		Path: ip.String(),
+	}
+}
```

Comment:
This will not be required once the dial target in the test is escaped (as per my other comment). And `MustParseURL` should only call `url.Parse()` and not do other things, so that it will remain close to what is being done in `grpc.ParseTarget`.

Reply:
Done - escaping the `%` did the trick.

I also updated documentation for `MustParseURL` and stripped the leading "/" from `URL.Opaque` to be consistent with [your previous comment](https://github.com/grpc/grpc-go/pull/5852#discussion_r1049140444).